api/v01/vrp.rb

```diff
@@ -546,28 +546,37 @@ def self.vrp_request_schedule(this)
               vrp_params = d_params[:points] ? d_params : d_params[:vrp]
               APIBase.dump_vrp_dir.write([api_key, vrp_params[:name], checksum].compact.join('_'), d_params.to_json) if OptimizerWrapper.config[:dump][:vrp]
 
-              APIBase.services(api_key)[:params_limit].merge(OptimizerWrapper.access[api_key][:params_limit] || {}).each{ |key, value|
+              APIBase.profile(api_key)[:params_limit].merge(OptimizerWrapper.access[api_key][:params_limit] || {}).each{ |key, value|
                 next if vrp_params[key].nil? || value.nil? || vrp_params[key].size <= value
 
                 error!({
                   status: 'Exceeded params limit',
-                  message: "Exceeded #{key} limit authorized for your account: #{value}. Please contact support or sales to increase limits."
-                }, 400)
+                  message: "Exceeded \"#{key}\" limit authorized for your account: #{value}. Please contact support or sales to increase limits."
+                }, 413)
               }
 
+              checksum = Digest::MD5.hexdigest Marshal.dump(params)
+              APIBase.dump_vrp_dir.write([api_key, vrp_params[:name], checksum].compact.join('_'), { vrp: vrp_params }.to_json) if OptimizerWrapper.config[:dump][:vrp]
+
               vrp = ::Models::Vrp.create(vrp_params)
+              count :submit_vrp, true, vrp.transactions
               if !vrp.valid? || vrp_params.nil? || vrp_params.keys.empty?
                 vrp.errors.add(:empty_file, message: 'JSON file is empty') if vrp_params.nil?
                 vrp.errors.add(:empty_vrp, message: 'VRP structure is empty') if vrp_params&.keys&.empty?
                 error!({ status: 'Model Validation Error', message: vrp.errors }, 400)
               else
-                ret = OptimizerWrapper.wrapper_vrp(api_key, APIBase.services(api_key), vrp, checksum)
+                ret = OptimizerWrapper.wrapper_vrp(api_key, APIBase.profile(api_key), vrp, checksum)
+
                 if ret.is_a?(String)
                   # present result, with: VrpResult
                   status 201
+                  count_incr :submit_vrp, transactions: vrp.transactions
```

Comment:
you cannot count transactions here, since you are not able to know if a result will be provided (if job crash, we don't charge any transaction). Instead, you should only store `count_key` and `transactions` in job (redis_resque) and call `count_incr` at the end of an aynchronuous job (at the end of `OptimizerWrapper::Job.perform`.